src/components/general/modals/prompt-login-modal.js

```diff
@@ -36,22 +36,24 @@ export default function PromptLoginModal() {
         aria-describedby="scroll-dialog-description"
       >
         <Title>
-          <LockIcon />
-          You are not logged in
+          {props.icon}
+          {props.titleText}
         </Title>
         <CloseButton aria-label="close" onClick={() => toggleModal(false)}>
           <CloseIcon />
         </CloseButton>
-        <Content dividers={scroll === 'paper'}>
-          You need to be logged in to AllClear to save pinned locations. Please login or create an account to continue.
-        </Content>
+        {!isNullOrUndefined(props.contentText) && ( 
+          <Content dividers={scroll === 'paper'}>
+            {props.contentText}
+          </Content>
+        )}
         <Actions disableSpacing={true}>
-          <PrimaryButton color={'primary'} variant={'contained'} onClick={() => history.push('/sign-in')}>
+          <LoginButton color={'primary'} variant={'contained'} onClick={() => history.push('/sign-in')}>
```

Comment:
eventually, you'd probably also want to pass in a config that sets up generic buttons and handlers, instead of housing this login logic within this generic component. for the purposes of this PR, you don't need to change this part though.